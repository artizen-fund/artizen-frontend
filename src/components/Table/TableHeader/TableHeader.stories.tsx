import TableHeader, { TableHeaderProps, Label } from './'
import { Button } from '@components'

export default {
  title: 'tables/TableHeader',
  component: TableHeader,
  argTypes: {
    children: {
      defaultValue: 'sample table',
      control: { type: 'text' },
    },
    ancillaryText: {
      defaultValue: 'more',
      control: { type: 'text' },
    },
  },
}

interface TableHeaderPropsWithAncillary extends TableHeaderProps {
  ancillaryText?: string
}

export const TableHeaderComponent = (props: TableHeaderProps) => <TableHeader {...props} />

export const TableHeaderWithButton = ({ ancillaryText, ...props }: TableHeaderPropsWithAncillary) => {
  const ancillary = <Button href="/">{ancillaryText}</Button>
  return <TableHeader {...props} {...{ ancillary }} />
}

export const TableHeaderWithSideLabel = ({ ancillaryText, ...props }: TableHeaderPropsWithAncillary) => {
  const ancillary = <Label>{ancillaryText}</Label>
  return <TableHeader {...props} {...{ ancillary }} />
}
