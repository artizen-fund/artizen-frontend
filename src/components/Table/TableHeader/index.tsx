import styled from 'styled-components'
import { TableProps } from '../'
import { breakpoint, palette } from '@theme'
import { rgba } from '@lib'

export type TableHeaderProps = Pick<TableProps, 'title' | 'sideItem'>

const TableHeader = ({ title, sideItem, ...props }: TableHeaderProps) => (
  <Wrapper {...props}>
    <Label>{title}</Label>
    {sideItem}
  </Wrapper>
)

const Wrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  padding-bottom: 12px;
  @media only screen and (min-width: ${breakpoint.desktop}px) {
    padding-bottom: 16px;
  }
`

export const Label = styled.div`
  font-size: 16px;
  line-height: 23px;
  color: ${rgba(palette.night)};
  @media (prefers-color-scheme: dark) {
    color: ${rgba(palette.moon)};
  }
`

export default TableHeader
