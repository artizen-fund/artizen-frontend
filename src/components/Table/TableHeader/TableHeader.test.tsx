import { render } from '@testing-library/react'
import TableHeader from './'

it('renders TableHeader unchanged', () => {
  const { container } = render(<TableHeader>Derp</TableHeader>)
  expect(container).toMatchSnapshot()
})
