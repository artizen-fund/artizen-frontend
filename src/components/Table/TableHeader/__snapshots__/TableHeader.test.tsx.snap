// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TableHeader unchanged 1`] = `
<div>
  <div
    class="sc-c6d4a71-0 cAjiXM"
  >
    <div
      class="sc-c6d4a71-1 dwQWaZ"
    />
  </div>
</div>
`;
